service/application/src/main/java/com/solace/maas/ep/event/management/agent/cli/EmaCommandLine.java

```diff
@@ -0,0 +1,210 @@
+        MessagingServiceEntity messagingServiceEntity = messagingServiceDelegateService.getMessagingServiceById(messagingServiceId);
+
+        ScanRequestBO scanRequestBO = new ScanRequestBO();
+        scanRequestBO.setMessagingServiceId(messagingServiceId);
+        scanRequestBO.setScanId(idGenerator.generateRandomUniqueId());
+        setScanType(messagingServiceEntity, scanRequestBO, messagingServiceId);
+        scanRequestBO.setDestinations(List.of("FILE_WRITER"));
+
+        log.info("Scan request [{}]: Received, request details: {}", scanRequestBO.getScanId(), scanRequestBO);
+        String scanId = scanManager.scan(scanRequestBO);
+
+        log.info("Scan request [{}]: Scan started.", scanId);
+        waitForScanToComplete(scanId);
+
+        if (isCompletedSuccessfully(scanId)) {
+            writeScanToZipFile(filePathAndName, scanId);
+        } else {
+            log.error("Scan request [{}]: Scan did not complete successfully.", scanId);
+        }
+    }
+
+    private boolean isCompletedSuccessfully(String scanId) {
+        List<ScanStatusEntity> statuses = scanStatusService.getScanStatuses(scanId);
+        boolean completedSuccessfully = false;
+        if (dataCollectionFileService.findAllByScanId(scanId).size() >=
+                statuses.size() && !anyScanStatusesInDesiredState("FAILED", statuses)) {
+            log.info("Scan request [{}]: Scan completed successfully.", scanId);
+            completedSuccessfully = true;
+        }
+        return completedSuccessfully;
+    }
+
+    private void setScanType(MessagingServiceEntity messagingServiceEntity, ScanRequestBO scanRequestBO, String messagingServiceId) {
+        switch (messagingServiceEntity.getType().toLowerCase()) {
+            case "solace":
+                scanRequestBO.setScanTypes(List.of("SOLACE_ALL"));
+                break;
+            case "kafka":
+                messagingServiceDelegateService.getMessagingServicesRelations(messagingServiceId).stream()
+                        .findFirst()
+                        .ifPresentOrElse(messagingServiceEntity1 -> scanRequestBO.setScanTypes(List.of("KAFKA_ALL", "CONFLUENT_SCHEMA_REGISTRY_SCHEMA")),
+                                () -> scanRequestBO.setScanTypes(List.of("KAFKA_ALL")));
+                break;
+            case "confluent_schema_registry":
+                scanRequestBO.setScanTypes(List.of("CONFLUENT_SCHEMA_REGISTRY_SCHEMA"));
+                break;
+            default:
+                throw new RuntimeException("Unsupported messaging service type: " + messagingServiceEntity.getType());
+        }
+    }
+
+    public void waitForScanToComplete(String scanId) throws InterruptedException {
+        AtomicBoolean scanCompleted = new AtomicBoolean(false);
+
+        // The hard timeout is set to 30 minutes to prevent the scan from running forever
+        long hardTimeoutMillis = System.currentTimeMillis() + 30 * 60 * 1000;
```

Comment:
This is the "paranoia" timeout just so the EMA doesn't run forever. It's currently set to 30 minutes since large configs can take a long time to scan.